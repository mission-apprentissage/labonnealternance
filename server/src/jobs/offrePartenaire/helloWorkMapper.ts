import { ObjectId } from "mongodb"
import { NIVEAUX_POUR_LBA, TRAINING_CONTRACT_TYPE, TRAINING_REMOTE_TYPE } from "shared/constants"
import dayjs from "shared/helpers/dayjs"
import { extensions } from "shared/helpers/zodHelpers/zodPrimitives"
import { JOBPARTNERS_LABEL } from "shared/models/jobsPartners.model"
import { IComputedJobsPartners } from "shared/models/jobsPartnersComputed.model "
import { z } from "zod"

export const ZHelloWorkJob = z
  .object({
    job_id: z.string(),
    contract_start_date: z.string().nullish(),
    contract: z.string(),
    remote: z.string().nullish(),
    contract_period_value: z.coerce.number().nullish(),
    contract_period_unit: z.string().nullish(),
    title: z.string(),
    description: z.string().nullish(),
    education: z.string().nullish(),
    profile: z.string().nullish(),
    code_rome: z.string().nullish(),
    publication_date: z.string().nullish(),
    siret: z.string().nullish(),
    company_title: z.string(),
    company_description: z.string().nullish(),
    address: z.string().nullish(),
    postal_code: z.string().nullish(),
    city: z.string().nullish(),
    geoloc: z.string().nullish(),
    url: extensions.url(),
  })
  .passthrough()

export type IHelloWorkJob = z.output<typeof ZHelloWorkJob>

const teletravailMapping: Record<string, TRAINING_REMOTE_TYPE> = {
  Complet: TRAINING_REMOTE_TYPE.remote,
  Partiel: TRAINING_REMOTE_TYPE.hybrid,
  Pas_teletravail: TRAINING_REMOTE_TYPE.onsite,
  Occasionnel: TRAINING_REMOTE_TYPE.hybrid,
}

function getDiplomaLevel(job: IHelloWorkJob): IComputedJobsPartners["offer_target_diploma"] {
  if (job.education == null) return null

  switch (job.education) {
    case "RJ/Qualif/BEP_CAP":
      return { european: "3", label: NIVEAUX_POUR_LBA["3 (CAP...)"] }
    case "RJ/Qualif/Employe_Operateur":
      return { european: "3", label: NIVEAUX_POUR_LBA["3 (CAP...)"] }
    case "RJ/Qualif/Technicien_B2":
      return { european: "5", label: NIVEAUX_POUR_LBA["5 (BTS, DEUST...)"] }
    case "RJ/Qualif/Agent_maitrise_B3":
      return { european: "6", label: NIVEAUX_POUR_LBA["6 (Licence, BUT...)"] }
    case "RJ/Qualif/Cadre_dirigeant":
      return { european: "7", label: NIVEAUX_POUR_LBA["7 (Master, titre ingénieur...)"] }
    case "RJ/Qualif/Ingenieur_B5":
      return { european: "7", label: NIVEAUX_POUR_LBA["7 (Master, titre ingénieur...)"] }
    default:
      return null
  }
}

export const helloWorkJobToJobsPartners = (job: IHelloWorkJob): IComputedJobsPartners => {
  const {
    contract,
    job_id,
    contract_start_date,
    remote,
    title,
    description,
    profile,
    code_rome,
    publication_date,
    siret,
    company_title,
    company_description,
    address,
    city,
    postal_code,
    geoloc,
    url,
  } = job
  const contractDuration: number | null = parseContractDuration(job)
  const { latitude, longitude } = geolocToLatLon(geoloc)
  const siretParsing = extensions.siret.safeParse(siret)
  const codeRomeParsing = extensions.romeCode().safeParse(code_rome)
  const urlParsing = extensions.url().safeParse(url)

  const partnerJob: IComputedJobsPartners = {
    _id: new ObjectId(),
    created_at: new Date(),
    partner_label: JOBPARTNERS_LABEL.HELLOWORK,
    partner_job_id: job_id,
    contract_start: parseDate(contract_start_date),
    contract_type: contract.toLowerCase() === "alternance" ? [TRAINING_CONTRACT_TYPE.APPRENTISSAGE, TRAINING_CONTRACT_TYPE.PROFESSIONNALISATION] : undefined,
    contract_remote: remote ? teletravailMapping[remote] ?? null : null,
    contract_duration: contractDuration,
    offer_title: title,
    offer_description: description && description.length >= 30 ? description : undefined,
    offer_target_diploma: getDiplomaLevel(job),
    offer_desired_skills: profile == null ? [] : [profile],
    offer_access_conditions: [],
    offer_to_be_acquired_skills: [],
    offer_rome_codes: codeRomeParsing.success ? [codeRomeParsing.data] : undefined,
    offer_creation: parseDate(publication_date),
    offer_expiration: null,
    offer_origin: null,
    offer_opening_count: 1,
    offer_multicast: false,
    workplace_siret: siretParsing.success ? siretParsing.data : null,
    workplace_name: company_title,
    workplace_description: company_description && company_description.length >= 30 ? company_description : null,
    workplace_size: null,
    workplace_website: null,
    workplace_address: {
      label: [address, postal_code, city].filter((x) => x).join(" "),
    },
    workplace_geopoint:
      latitude && longitude
        ? {
            type: "Point",
            coordinates: [longitude, latitude],
          }
        : undefined,
    apply_url: urlParsing.success ? urlParsing.data : null,
    errors: [],
    validated: false,
  }
  return partnerJob
}

const acceptedGeoLocRegex = /-?[0-9]{1,2}[.,][0-9]+,-?[0-9]{1,3}[.,][0-9]+/

const geolocToLatLon = (geoloc: string | null | undefined) => {
  if (!geoloc || !acceptedGeoLocRegex.test(geoloc)) return {}
  const parts = geoloc.split(",")
  if (parts.length !== 2 && parts.length !== 4) return {}
  const parsedParts = parts.map(parseFloat)
  if (parsedParts.some(isNaN)) return {}
  if (parts.length === 2) {
    return { latitude: parsedParts[0], longitude: parsedParts[1] }
  } else if (parts.length === 4) {
    const latitude = parseFloat(`${parsedParts[0]}.${parsedParts[1]}`)
    const longitude = parseFloat(`${parsedParts[2]}.${parsedParts[3]}`)
    return { latitude, longitude }
  } else {
    throw new Error("inattendu")
  }
}

const parseContractDuration = ({ contract_period_unit, contract_period_value }: { contract_period_unit?: string | null; contract_period_value?: number | null }): number | null => {
  if (!contract_period_unit || !contract_period_value) {
    return null
  }
  switch (contract_period_unit.toLowerCase()) {
    case "year":
      return contract_period_value * 12
    case "month":
      return contract_period_value
    case "week":
      return Math.ceil((contract_period_value * 7) / (365 / 12))
  }
  return null
}

const parseDate = (dateStr: string | null | undefined) => {
  if (!dateStr) {
    return null
  }
  return dayjs.tz(dateStr).toDate()
}
