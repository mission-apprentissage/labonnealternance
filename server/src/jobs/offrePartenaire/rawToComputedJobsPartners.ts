import { internal } from "@hapi/boom"
import { oleoduc, writeData } from "oleoduc"
import { z } from "shared/helpers/zodWithOpenApi"
import { JOBPARTNERS_LABEL, ZJobsPartnersOfferPrivate } from "shared/models/jobsPartners.model"
import { IComputedJobsPartners } from "shared/models/jobsPartnersComputed.model "
import { CollectionName } from "shared/models/models"
import { AnyZodObject } from "zod"

import { logger } from "@/common/logger"
import { getDbCollection } from "@/common/utils/mongodbUtils"
import { sentryCaptureException } from "@/common/utils/sentryUtils"

export const rawToComputedJobsPartners = async <ZodInput extends AnyZodObject>({
  collectionSource,
  mapper,
  zodInput,
  partnerLabel,
  documentJobRoot,
}: {
  collectionSource: CollectionName
  zodInput: ZodInput
  mapper: (raw: z.output<ZodInput>) => IComputedJobsPartners
  partnerLabel: JOBPARTNERS_LABEL
  documentJobRoot?: string
}) => {
  logger.info(`début d'import dans computed_jobs_partners pour partner=${partnerLabel}`)
  const deletedCount = await getDbCollection("computed_jobs_partners").countDocuments({ partner_label: partnerLabel })
  logger.info(`suppression de ${deletedCount} documents dans computed_jobs_partners pour partner=${partnerLabel}`)
  await getDbCollection("computed_jobs_partners").deleteMany({ partner_label: partnerLabel })
  const counters = { total: 0, success: 0, error: 0 }
  await oleoduc(
    getDbCollection(collectionSource).find({}).stream(),
    writeData(
      async (document) => {
        counters.total++
        try {
          const rawJob = documentJobRoot ? document[documentJobRoot] : document
          const parsedDocument = zodInput.parse(rawJob)
          const computedJobPartner = mapper(parsedDocument)
          await getDbCollection("computed_jobs_partners").insertOne({
            ...computedJobPartner,
            partner: partnerLabel,
            validated: ZJobsPartnersOfferPrivate.safeParse(computedJobPartner).success,
          })
          counters.success++
        } catch (err) {
          counters.error++
          const newError = internal(`error converting raw job to partner job for partner=${partnerLabel}, id=${document._id}`)
          logger.error(newError.message, err)
          newError.cause = err
          sentryCaptureException(newError)
        }
      },
      { parallel: 10 }
    )
  )
  logger.info(`import dans computed_jobs_partners pour partner=${partnerLabel} terminé`, counters)
}
