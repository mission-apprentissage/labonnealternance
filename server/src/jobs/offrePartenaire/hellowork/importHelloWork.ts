import { JOBPARTNERS_LABEL } from "shared/models/jobsPartners.model"
import rawHelloWorkModel from "shared/models/rawHelloWork.model"

import config from "../../../config"
import { importFromStreamInXml } from "../importFromStreamInXml"
import { importFromUrlInXml } from "../importFromUrlInXml"
import { rawToComputedJobsPartners } from "../rawToComputedJobsPartners"

import { helloWorkJobToJobsPartners, ZHelloWorkJob } from "./helloWorkMapper"

const rawCollectionName = rawHelloWorkModel.collectionName
const offerXmlTag = "job"

export const importHelloWorkRaw = async (sourceStream?: NodeJS.ReadableStream) => {
  if (sourceStream) {
    await importFromStreamInXml({ destinationCollection: rawCollectionName, offerXmlTag, stream: sourceStream, partnerLabel: JOBPARTNERS_LABEL.HELLOWORK })
  } else {
    await importFromUrlInXml({ destinationCollection: rawCollectionName, url: config.helloworkUrl, offerXmlTag, partnerLabel: JOBPARTNERS_LABEL.HELLOWORK })
  }
}

export const importHelloWorkToComputed = async () => {
  await rawToComputedJobsPartners({
    collectionSource: rawCollectionName,
    partnerLabel: JOBPARTNERS_LABEL.HELLOWORK,
    zodInput: ZHelloWorkJob,
    mapper: helloWorkJobToJobsPartners,
    documentJobRoot: offerXmlTag,
  })
}
