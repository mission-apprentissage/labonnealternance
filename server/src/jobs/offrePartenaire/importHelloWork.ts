import { JOBPARTNERS_LABEL } from "shared/models/jobsPartners.model"

import { helloWorkJobToJobsPartners, ZHelloWorkJob } from "./helloWorkMapper"
import { importFromStreamInXml } from "./importFromStreamInXml"
import { importFromUrlInXml } from "./importFromUrlInXml"
import { rawToComputedJobsPartners } from "./rawToComputedJobsPartners"

export const importHelloWork = async (sourceStream?: NodeJS.ReadableStream) => {
  if (sourceStream) {
    await importFromStreamInXml({ destinationCollection: "raw_hellowork", offerXmlTag: "job", stream: sourceStream })
  } else {
    await importFromUrlInXml({ destinationCollection: "raw_hellowork", url: "plop", offerXmlTag: "job" })
  }
  await rawToComputedJobsPartners({
    collectionSource: "raw_hellowork",
    partnerLabel: JOBPARTNERS_LABEL.HELLOWORK,
    zodInput: ZHelloWorkJob,
    mapper: helloWorkJobToJobsPartners,
    documentJobRoot: "job",
  })
}
