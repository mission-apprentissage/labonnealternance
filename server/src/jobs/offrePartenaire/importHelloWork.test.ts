import fs from "node:fs"

import { useMongo } from "@tests/utils/mongo.test.utils"
import { JOBPARTNERS_LABEL } from "shared/models/jobsPartners.model"
import { describe, expect, it } from "vitest"

import { getDbCollection } from "@/common/utils/mongodbUtils"

import { importHelloWork } from "./importHelloWork"

describe("importHelloWork", () => {
  useMongo()

  it("should test the whole import", async () => {
    const fileStream = fs.createReadStream("server/src/jobs/offrePartenaire/importHelloWork.test.input.xml")
    await importHelloWork(fileStream)
    expect(await getDbCollection("raw_hellowork").countDocuments({})).toBe(5)
    const jobs = (
      await getDbCollection("computed_jobs_partners")
        .find({ partner_label: JOBPARTNERS_LABEL.HELLOWORK }, { projection: { _id: 0, created_at: 0 } })
        .sort({ partner_id: 1 })
        .toArray()
    ).map((job) => ({ ...job, offer_creation_date: job.offer_creation_date?.toISOString() }))
    const expectedJson = JSON.parse(await fs.readFileSync("server/src/jobs/offrePartenaire/importHelloWork.test.expected.json", "utf-8"))
    expect(jobs).toEqual(expectedJson)
  })
})
