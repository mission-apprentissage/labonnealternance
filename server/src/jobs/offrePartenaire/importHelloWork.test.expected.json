[
  {
    "partner_label": "Hellowork",
    "partner_id": "68087320",
    "contract_start": null,
    "contract_type": null,
    "contract_remote": "Présentiel",
    "contract_duration": null,
    "offer_title": "Manager en Stratégie Management et Organisation H/F",
    "offer_description": "Vous développerez l'innovation pour créer les conditions favorables à l'émergence de nouveaux usages dans nos pratiques de conseil et dans le cadre des projets clients. Vous participerez au processus de recrutement et à la montée en compétences des consultants.\n\nVous contribuez au développement commercial en concevant des nouvelles offres, développant le portefeuille client existant, répondant aux appels d'offres clients. Vous participez aux actions de notoriété et de communication du cabinet : études de marché, publications d'articles ou d'infographies.",
    "offer_diploma_level_label": "Master, titre ingénieur, autres formations niveau (Bac+5)",
    "offer_desired_skills": "-   Expert du conseil en management et du secteur banque ou assurance.\n-   Esprit entrepreneurial et appétence commerciale\n-   Goût pour le numérique et l'innovation.\n-   Aisance à assurer un pilotage de projet alternant présentiel et distanciel.\n-   Créatif ayant la capacité à mettre en oeuvre ses idées\n-   Leadership et Capacité à nouer des relations de confiance avec les clients.\n-   Esprit d'équipe et d'initiative.\n-   Conscient des enjeux collectifs.\n-   Pédagogue qui aime transmettre ses compétences.\n-   Anglais : lu, écrit, parlé.",
    "offer_access_condition": null,
    "offer_acquired_skills": null,
    "offer_rome_code": null,
    "offer_creation_date": "2024-01-30T22:21:39.000Z",
    "offer_expiration_date": null,
    "offer_origin": null,
    "offer_count": 1,
    "offer_multicast": false,
    "workplace_siret": null,
    "workplace_name": "Software Consulting",
    "workplace_description": "Software Consulting est un cabinet de conseil en stratégie, management et en organisation indépendant et spécialisé dans les services financiers : la banque et l'assurance.",
    "workplace_size": null,
    "workplace_website": null,
    "workplace_raison_sociale": null,
    "workplace_enseigne": null,
    "workplace_address": "94300 VINCENNES",
    "workplace_geopoint": { "type": "Point", "coordinates": [48.123456789, 2.123456789] },
    "apply_url": "https://r.holeest.com/redirect?poc=2&op=4368162035&o=0&from=https%3A%2F%2Fwww.hellowork.com%2Ffr-fr%2Femplois%2F41990977.html%3Futm_medium%3DMetamoteurs-cpc%26utm_source%3Dexample%26utm_campaign%3Dclassique%2B-%2Bmarketing%252ccommunication%252cgraphisme%252caudiovisuel%252cpub%26utm_term%3D41990977%26utm_content%3Dfluxexample",
    "errors": [],
    "validated": false
  },
  {
    "partner_label": "Hellowork",
    "partner_id": "68090104",
    "contract_start": null,
    "contract_type": null,
    "contract_remote": null,
    "contract_duration": null,
    "offer_title": "Chargé de Communication Btob Laval 53 H/F",
    "offer_description": "Votre futur poste\n\nVenez participer à la création et à l'élaboration des actions et outils de communication BtoB au sein de notre Direction Appro lait basée à Laval (53).\n\nAfin de coordonner et mettre en oeuvre les actions et outils de communication de la Direction Approvisionnement Lait France en relation avec la direction communication, vos principales missions seront :\n-  Déployer la stratégie digitale et de contenu sur le site web des producteurs laitiers en cohérence avec la ligne éditoriale (Définir un planning éditorial, les contenus des publications, les publier et animer ainsi qu'assurer le reporting des actions\n-  Déployer des solutions facilitant l'accès à l'information aux producteurs (portail web producteur, pushmail) dans le but d'ancrer les usages digitaux dans le comportement des producteurs, d'accroître le trafic sur le site et d'attirer de nouveaux utilisateurs.\n-  Réaliser des supports d'aide à la vente (plaquettes) en coordination avec les responsables commerciaux, et assurer la création et la diffusion d'outils webmarketing.\n-  Accompagner les équipes qualité & RSE sur la conception d'outils de communication à destination de nos équipes techniques et de nos producteurs\n-  Mettre en oeuvre la présence de Lactalis France sur les salons et événements professionnels, organiser des événements à destination des clients\n-  Concevoir les outils de communication print en lien avec le studio graphique (magazine, plaquettes, affiche, encart web)\n\nCe poste est à pourvoir en CDI\n\nAccomplissez-vous avec engagement et plaisir au quotidien !\n\nPourquoi nous rejoindre ?\n\nParce que la promotion interne est au coeur de notre modèle, vous bénéficierez d'un accompagnement personnalisé et adapté à vos besoins pour construire votre projet professionnel avec des perspectives d'évolution.\n\nVous pourrez aborder l'avenir en toute confiance avec notamment une rémunération reconnaissant votre engagement : salaire fixe sur 13 mois, prime d'objectifs, intéressement, participation et épargne salariale attractive. Vous bénéficierez par ailleurs d'une mutuelle et des avantages de votre CSE.\n\nVous recherchez une ville à taille humaine bénéficiant d'un cadre de vie verdoyant ? Venez découvrir Laval, à seulement 1h10 de Paris et à proximité de grandes villes comme Rennes et Angers !\n\nVenez évoluer au sein d'équipes passionnées !\n\nVos atouts\n\nVous êtes issu d'une formation de type Master en Marketing, communication ou spécialité agri/agro, et vous justifiez idéalement d'une première expérience dans la communication. L'appétence pour le milieu AGRI/AGRO est également primordiale pour ce poste\n\nVous maîtrisez différentes techniques de rédaction (écriture web, print), de publication on line et des outils d'analyses de trafic\n\nEt une bonne connaissance des logiciels de PAO (QuarkXPress), de retouche photo (Photoshop, InDesign, Illustrator) et en CMS (Content Management System)\n\nEnfin vous avez une bonne culture des différents procédés de mise en oeuvre des actions et des techniques nécessaires afin de contrôler la qualité des réalisations (brochures, plaquettes, site internet, réseaux sociaux) et de tenir les délais de production\n\nEt surtout, vous êtes doté d'un fort esprit de synthèse et faites preuve d'un excellent relationnel, alors ce poste est fait pour vous !\n\nRévélez votre potentiel dans un environnement « apprenant » !\n\nCe poste correspond à votre projet ?  Alors postulez !\n\nAprès un premier échange téléphonique, vous rencontrerez votre futur Manager et un Responsable Ressources Humaines.\n\nRejoignez-nous et venez goûter à une expérience unique !",
    "offer_diploma_level_label": "Master, titre ingénieur, autres formations niveau (Bac+5)",
    "offer_desired_skills": null,
    "offer_access_condition": null,
    "offer_acquired_skills": null,
    "offer_rome_code": null,
    "offer_creation_date": "2024-01-30T22:11:59.000Z",
    "offer_expiration_date": null,
    "offer_origin": null,
    "offer_count": 1,
    "offer_multicast": false,
    "workplace_siret": "33114255400012",
    "workplace_name": "Lactalis",
    "workplace_description": "Vous aussi venez vivre votre #LactalisExperience !\nFondé en 1933 par André Besnier à Laval, le Groupe Lactalis est une entreprise familiale devenue en trois générations le leader mondial des produits laitiers (250 sites de production dans 43 pays - 85 000 collaborateurs)",
    "workplace_size": null,
    "workplace_website": null,
    "workplace_raison_sociale": null,
    "workplace_enseigne": null,
    "workplace_address": "53000 LAVAL",
    "workplace_geopoint": { "type": "Point", "coordinates": [48.123456789, -0.123456789] },
    "apply_url": "https://r.holeest.com/redirect?poc=2&op=4368162036&o=0&from=https%3A%2F%2Fwww.hellowork.com%2Ffr-fr%2Femplois%2F41995446.html%3Futm_medium%3DMetamoteurs-cpc%26utm_source%3Dexample%26utm_campaign%3Dclassique%2B-%2Bmarketing%252ccommunication%252cgraphisme%252caudiovisuel%252cpub%26utm_term%3D41995446%26utm_content%3Dfluxexample",
    "errors": [],
    "validated": false
  },
  {
    "partner_label": "Hellowork",
    "partner_id": "68110296",
    "contract_start": null,
    "contract_type": null,
    "contract_remote": "Hybride",
    "contract_duration": null,
    "offer_title": "Directeur Marketing H/F",
    "offer_description": "Nous recherchons notre futur(e) Directeur Marketing.\n\nVos futures missions : si vous les acceptez\n\nMission 1 : Élaboration et Développement de la Stratégie Marketing\n\nEn tant que Directeur Marketing du Comptoir Général de Robinetterie, vous serez le moteur de notre succès en élaborant et en mettant en oeuvre notre stratégie marketing. Vos responsabilités clés incluront :\n-  Concevoir et mettre en oeuvre une stratégie marketing en alignement avec les objectifs de l'entreprise.\n-  Contribuer à la stratégie omnicanale de l'entreprise, tant en ligne qu'hors ligne.\n-  Superviser le plan de déploiement et d'animation du canal de vente en ligne.\n-  Élaborer et superviser le plan marketing opérationnel de l'entreprise.\n-  Concevoir des plans marketing, y compris la définition des segments de marché, la tarification, la promotion et la distribution.\n-  Surveiller les tendances du marché, la concurrence et les besoins des clients pour adapter la stratégie.\n-  Diriger de nouveaux projets digitaux.\n-  Travailler en étroite collaboration avec les équipes de vente pour développer des outils et des stratégies de vente efficaces.\n-  Proposer des plans d'évolution des produits et services.\n\nMission 2 : Définition de la Stratégie de Communication Interne et Externe\n\nVous serez également chargé de définir la stratégie de communication interne et externe pour renforcer la visibilité de l'entreprise. Vos activités comprendront :\n-  Participer à la définition de la stratégie de communication interne et externe pour atteindre les publics cibles.\n-  Élaborer des plans de communication internes et externes.\n-  Mettre en place des indicateurs de performance (KPI) pour évaluer l'efficacité des actions de communication.\n-  Assurer la cohérence des actions de communication avec la stratégie globale.\n-  Gérer le budget de communication et effectuer un reporting à la direction générale.\n-  Optimiser les processus pour améliorer le service de communication.\n\nMission 3 : Management des Équipes Marketing et Communication\n\nEn tant que leader, vous dirigerez les équipes marketing et communication, coordonnerez leurs activités et encouragerez le développement professionnel. Vos activités comprendront :\n-  Superviser les équipes de marketing et de communication.\n-  Recruter et animer les équipes marketing.\n-  Coordonner les collaborations entre les équipes marketing et les équipes commerciales.\n-  Diriger et motiver l'équipe pour atteindre les objectifs fixés.\n-  Évaluer et contribuer au développement professionnel de vos collaborateurs.\n-  Agir pour le développement et l'épanouissement de votre équipe.",
    "offer_diploma_level_label": "Master, titre ingénieur, autres formations niveau (Bac+5)",
    "offer_desired_skills": "Exigences du poste :\n-   Expérience solide en marketing, de préférence dans l'industrie de la robinetterie.\n-   Connaissance des produits de robinetterie et du secteur.\n-   Capacité à élaborer et exécuter des stratégies marketing efficaces.\n-   Excellentes compétences en gestion d'équipe et en coordination.\n-   Fortes compétences en communication, à la fois écrite et orale.\n-   Diplôme en marketing, communication ou domaine connexe (un atout).",
    "offer_access_condition": null,
    "offer_acquired_skills": null,
    "offer_rome_code": null,
    "offer_creation_date": "2024-01-31T22:09:13.000Z",
    "offer_expiration_date": null,
    "offer_origin": null,
    "offer_count": 1,
    "offer_multicast": false,
    "workplace_siret": "31882776300042",
    "workplace_name": "CGR",
    "workplace_description": "Distributeur spécialisé en robinetterie et accessoires depuis plus de 40 ans, nous accompagnons les professionnels du Génie Climatique, du Logement et de l'Industrie dans la réussite de leurs projets.\n\nPour ce faire, nous proposons une gamme élargie de solutions et de services associée à une logistique de pointe et à l'expertise de nos 300 collaborateurs.\n\nNos valeurs : Respect - Bien-être - Détermination - Innovation\n\nEn constante évolution pour satisfaire nos clients, nous recherchons de nouveaux talents passionnés et engagés qui souhaitent s'investir pour révolutionner durablement la distribution.\n\nPrêt à rejoindre la Team CGR ?",
    "workplace_size": null,
    "workplace_website": null,
    "workplace_raison_sociale": null,
    "workplace_enseigne": null,
    "workplace_address": "51430 TINQUEUX",
    "workplace_geopoint": { "type": "Point", "coordinates": [49.249622311, 3.989532671] },
    "apply_url": "https://r.holeest.com/redirect?poc=2&op=4368162037&o=0&from=https%3A%2F%2Fwww.hellowork.com%2Ffr-fr%2Femplois%2F42014902.html%3Futm_medium%3DMetamoteurs-cpc%26utm_source%3Dexample%26utm_campaign%3Dclassique%2B-%2Bmarketing%252ccommunication%252cgraphisme%252caudiovisuel%252cpub%26utm_term%3D42014902%26utm_content%3Dfluxexample",
    "errors": [],
    "validated": false
  },
  {
    "partner_label": "Hellowork",
    "partner_id": "68111676",
    "contract_start": null,
    "contract_type": null,
    "contract_remote": "Présentiel",
    "contract_duration": null,
    "offer_title": "Ingénieur Intégration Software H/F",
    "offer_description": "La révolution robotique portée par Exotec ne fait que commencer, vous en êtes ?\n\nResponsibilities\n\n-  Développer des programmes logiciels et des interfaces entre les produits Exotec et les écosystèmes logiciels des clients\n-  Rédiger les spécifications pour tous les développements de logiciels des projets\n-  Analyser les besoins et l'architecture logicielle des clients et faire des propositions techniques optimisées\n-  Rédiger l'analyse fonctionnelle décrivant les opérations logicielles sur les systèmes\n-  Assurer une passation des fonctionnalités logicielles développées aux équipes sur site\n-  Soutenir (sur place ou à distance) les équipes sur site si nécessaire\n\nRequirements\n\n-  Vous avez une formation orientée informatique acquise en école d'ingénieur ou avec un cursus Bac +4/+5\n-  Vous avez une première expérience dans le développement et/ou l'intégration de logiciels de type WMS (WMS, WCS, TMS) dans le secteur logistique\n-  Vous maitrisez les compétences techniques suivantes : SQL / PostgreSQL, JavaScript, Web développement\n-  Vous aimez le challenge, la diversité et la résolution de problématiques techniques innovantes\n-  Vous êtes dynamique, rigoureux(se) et aimez travailler en équipe sur des missions variées\n-  Vous avez un niveau d'anglais opérationnel, à l'oral comme à l'écrit\n\nBenefits\n\n-  Couverture mutuelle et prévoyance santé compétitive\n-  Primes collectives et attribution de BSPCE\n-  Politique famille avantageuse\n-  Programme de mobilité interne et internationale\n-  Nombreuses opportunités de formation et de développement\n\nChez Exotec, nous garantissons l'égalité des chances dans notre processus de recrutement. L'ensemble des candidatures reçues sont étudiées indépendamment de l'âge, du genre, de l'origine, de la religion, de la couleur de peau, de la nationalité, du sexe, du handicap, de l'orientation sexuelle ou de toute autre distinction protégée par la loi. Nous mettons en place un environnement de travail inclusif et respectueux de toutes les différences. En rejoignant le Pacte Parité, Exotec s'engage pour un écosystème French Tech plus paritaire",
    "offer_diploma_level_label": "Master, titre ingénieur, autres formations niveau (Bac+5)",
    "offer_desired_skills": null,
    "offer_access_condition": null,
    "offer_acquired_skills": null,
    "offer_rome_code": null,
    "offer_creation_date": "2024-01-31T22:11:11.000Z",
    "offer_expiration_date": null,
    "offer_origin": null,
    "offer_count": 1,
    "offer_multicast": false,
    "workplace_siret": "81287874200038",
    "workplace_name": "Exotec",
    "workplace_description": "Chez Exotec, nous mettons l'excellence technologique au service de la redéfinition des relations entre humains et robots. A travers le monde, nos solutions révolutionnent la façon dont nos clients délivrent leurs produits aux consommateurs finaux. Nous contribuons au succès des plus grandes marques du commerce et de l'industrie, tout en améliorant les conditions de travail de leurs salariés.\n\nPar l'alliance de l'intelligence artificielle et d'un hardware performant, nos robots sont désormais déployés dans le monde entier et leur succès a fait de nous la première licorne industrielle française.\n\nRejoindre Exotec, c'est l'opportunité de donner du sens à vos compétences. Grandissez avec plus de 600 ExoPeople dans le monde entier pour faire de vos idées, des réalités.",
    "workplace_size": null,
    "workplace_website": null,
    "workplace_raison_sociale": null,
    "workplace_enseigne": null,
    "workplace_address": "59160 LILLE",
    "workplace_geopoint": { "type": "Point", "coordinates": [50.6317183168, 3.04783272312] },
    "apply_url": "https://r.holeest.com/redirect?poc=2&op=4368162038&o=0&from=https%3A%2F%2Fwww.hellowork.com%2Ffr-fr%2Femplois%2F42016309.html%3Futm_medium%3DMetamoteurs-cpc%26utm_source%3Dexample%26utm_campaign%3Dclassique%2B-%2Bmarketing%252ccommunication%252cgraphisme%252caudiovisuel%252cpub%26utm_term%3D42016309%26utm_content%3Dfluxexample",
    "errors": [],
    "validated": false
  },

  {
    "partner_label": "Hellowork",
    "partner_id": "68113693",
    "contract_start": null,
    "contract_type": null,
    "contract_remote": "Présentiel",
    "contract_duration": null,
    "offer_title": "Chef de Projet Photo H/F",
    "offer_description": "Prendre en charge plusieurs projets de développement de produits sous la responsabilité d'un responsable de gamme.\n\nSuivre le cahier des charges Marketing pour définir et figer les spécifications du produit.\n\nTravailler avec les achats pour définir un prix produit et le garantir - make or buy.\n\nEtablir le besoin du projet en ressources, budget et son planning et veiller à les respecter.\n\nRendre compte au responsable de gamme et à la direction des avancées, besoins et risques du projet.\n\nPiloter les actions pluridisciplinaires (électronique, mécanique, logiciel, validation) des équipes internes et fournisseurs qui conditionnent la fabrication industrielle d'un produit, de la conception à la mise en production. Mettre en place les outils de gestion liés aux produits (documents techniques, structures informatiques, bases de données) et tenir à jour ces données (plans, nomenclatures, procédures).\n\nVeiller à la qualification du produit en collaboration avec les équipes validation et également les organismes de certifications extérieurs.\n\nVeiller à la mise en oeuvre des livrables attendus à chaque jalon (spécification, plans, nomenclatures, schémas électroniques, logiciel, procédures).\n\nÊtre le premier contact des partenaires, fournisseurs et sous-traitants pour toutes les questions liées à l'industrialisation. Assurer les actions nécessaires à la maintenance des produits en vie série (création / validation kits SAV, gestion obsolescence composants, correction bugs, programme de réduction des coûts)\n\nAssurer les actions nécessaires à la maintenance des produits en vie série (création / validation kits SAV, gestion obsolescence composants, correction bugs, programme de réduction des coûts).",
    "offer_diploma_level_label": "Master, titre ingénieur, autres formations niveau (Bac+5)",
    "offer_desired_skills": "FORMATION - Formation d'Ingénieur généraliste ou industriel ou Mécanique,\n\nElectronique.\n\nEXPERIENCE - 3 à 5 ans dans le domaine industriel, plus\n\nparticulièrement dans une entreprise de conception, production, commercialisation\n\nde produits assemblés.\n\nQUALITES - Rigueur, capacité à s'organiser, leadership, communication, pragmatisme, autonomie\n\nLANGUES - Anglais courant.\n\nRef : 1jvie4rokd",
    "offer_access_condition": null,
    "offer_acquired_skills": null,
    "offer_rome_code": null,
    "offer_creation_date": "2024-01-31T22:22:17.000Z",
    "offer_expiration_date": null,
    "offer_origin": null,
    "offer_count": 1,
    "offer_multicast": false,
    "workplace_siret": "81986335800019",
    "workplace_name": "ME GROUP",
    "workplace_description": "ME GROUP (Photomaton) développe et commercialise des distributeurs automatiques en libre-service à valeur ajoutée. Aujourd'hui, nous sommes un acteur majeur dans des secteurs aussi divers que la photographie, l'alimentation et les laveries automatiques.",
    "workplace_size": null,
    "workplace_website": null,
    "workplace_raison_sociale": null,
    "workplace_enseigne": null,
    "workplace_address": "Photo-Me,Échirolles,38130 38130 ECHIROLLES",
    "workplace_geopoint": { "type": "Point", "coordinates": [45.1471647181, 5.71535600177] },
    "apply_url": "https://r.holeest.com/redirect?poc=2&op=4368162039&o=0&from=https%3A%2F%2Fwww.hellowork.com%2Ffr-fr%2Femplois%2F42018480.html%3Futm_medium%3DMetamoteurs-cpc%26utm_source%3Dexample%26utm_campaign%3Dclassique%2B-%2Bmarketing%252ccommunication%252cgraphisme%252caudiovisuel%252cpub%26utm_term%3D42018480%26utm_content%3Dfluxexample",
    "errors": [],
    "validated": false
  }
]
